From 0252ab738e2a1b43d71ccab35725f7fede281225 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 20 Nov 2024 06:49:52 +0200
Subject: [PATCH] [SW-0]: [test] agent signing with custom keys

Test patch to perform signing of image pointer with custom
keys

Keys can be copied from here:
cp board/habanalabs/gaudi2/key/cust* output/gaudi3_arc/images/
cp board/habanalabs/gaudi2/key/cust* output/gaudi3/images/

Run command 'make gaudi3-asic-build' t perform signing

in order to sign with different key index, change var KEY_NUM

Change-Id: I707afacb1ac77eb85881a00035623af313539307
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index eb4a2c7..399cb9e 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -100,7 +100,8 @@
 	OUT_NAME=$3
 
 	TMPNAME=${STAGE_DIR}/${KEY_NAME}-${PROP_NAME}.txt
-	fdtget ${BINS_PROD_DIR}/${KEY_NAME}.dtb /signature/key-${KEY_NAME} rsa,${PROP_NAME} -t bx > ${TMPNAME}
+	#fdtget ${BINS_PROD_DIR}/${KEY_NAME}.dtb /signature/key-${KEY_NAME} rsa,${PROP_NAME} -t bx > ${TMPNAME}
+	fdtget ${BINS_PROD_DIR}/${KEY_NAME}.dtb /signature/key-hl_rsa3072 rsa,${PROP_NAME} -t bx > ${TMPNAME}
 	sed 's/\s\+/\n/g' ${TMPNAME} > ${TMPNAME}.tmp
 	sed -e 's/\b[0-9]\b/0&/' -e 's/\b[a-f]\b/0&/' ${TMPNAME}.tmp  > ${TMPNAME}.tmp.le
 	xxd -r -p ${TMPNAME}.tmp.le > ${OUT_NAME}
@@ -267,6 +268,7 @@
 	local input=$(basename $1)
 	output=${input_full_path}.tmp
 	hash_output=${STAGE_DIR}/img-hash-${input}.sha384
+	LOCAL_FILE=/home/vmaira/tpm/
 
 	input_size=$(stat -c %s "$input_full_path")
 	local pub_key_size=784
@@ -278,7 +280,9 @@
 	count=$(( $input_size - $truncate_bytes ))
 	dd if="$input_full_path" of="$output" bs=1 count=${count}
 	openssl dgst -sha384 -binary ${output} > ${hash_output}
-	rm -rf ${output}
+
+    cp ${output} ${LOCAL_FILE}
+	#rm -rf ${output}
 }
 
 function prep_img_desc_with_crc {
@@ -457,12 +461,22 @@
 	print_image_version $img_name
 }
 
+
+KEYS_DIR=/home/vmaira/trees/npu-stack/embedded/buildroot-external/board/habanalabs/gaudi2/key
+function hl_copy_keys
+{
+        FILEPATH=$1
+
+        cp ${KEYS_DIR}/${KEY_NAME}.key ${FILEPATH}
+        cp ${KEYS_DIR}/${KEY_NAME}.dtb ${FILEPATH}
+}
+
 function prep_image_ptrs_common_v5 {
 	local img_file=$1
 	local img_ptrs_sz=32
 	local img_desc_sz=348
 	local out_file=$1
-	local KEY_NUM=0
+	local KEY_NUM=1
 	local EROM_IMG=$2
 
 	# Prepare the NEXT and MODIFIABLE for backward compatibility
@@ -553,6 +567,8 @@
 		KEY_NAME=$(echo -n "cust${KEY_NUM}_${KEY_BASE_NAME}")
 	fi
 
+	hl_copy_keys ${BINS_PROD_DIR}
+
 	#Extract Key Properties and store them in a bin file
 	hl_get_key_props ${KEY_NUM}
 
